File: codechef/aug_long/guddu.py
t = int(input())
for i in range(t):
	n = int(input())
	ar = input().split()
	for i in range(len(ar)):
		ar[i] = int(ar[i])
	pref = [0]
	for i in range(n):
		pref.append(pref[i]^ar[i])
	b = dict()
	for i in range(n+1):
		if pref[i] not in b:
			b[pref[i]] = []
		b[pref[i]].append(i);
	ans=0
	for p in b:
		size = len(b[p])
		sum = [0 for i in range(size)]
		sum[size-1] = b[p][-1];
		for j in range(size-2,0,-1):
			sum[j] = sum[j+1] + b[p][j];
		contrib = 0;
		for j in range(size-1):
			contrib -= (size-1-j)*b[p][j];
			contrib += sum[j+1];
			contrib -= (size-1-j);
		ans += contrib;
	print(ans)

File: codechef/aug_long/guddu.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
	int t;
	cin >> t;
	for (int i = 0; i < t; i++) {
		int n;
		cin >> n;
		int ar[n];
		for (int i = 0; i < n; i++) {
			cin >> ar[i];
		}
		int pref[n+1];
		pref[0] = 0;
		//cout << "pref: " << pref[0] << ' ';
		for (int i = 0; i < n; i++) {
			pref[i+1] = pref[i]^ar[i];
			//cout << pref[i+1] << ' ';
		}
		//cout << endl;
		map<int, vector<int> > b;
		for (int i = 0; i < n+1; i++) {
			b[pref[i]].push_back(i);
		}
		int ans=0;
		for (auto p: b) {
			//cout << "in bucket " << p.first << endl;
			int size = p.second.size();
			int sum[size];
			sum[size-1] = p.second.back();
			//cout << "sum: " << sum[size-1] << ' ';
			for (int j = size-2; j >= 0; j--) {
				sum[j] = sum[j+1] + p.second[j];
				//cout << sum[j] << ' ';
			}
			//cout << endl;
			int contrib = 0;
			for (int j = 0; j < size-1; j++) {
				contrib -= (size-1-j)*p.second[j];
				contrib += sum[j+1];
				contrib -= (size-1-j);
			}
			//cout << "contrib is " << contrib << endl;
			ans += contrib;
		}
		cout << ans << endl;
	}
	return 0;
}
